src/sql/src/statements.rs

```diff
@@ -197,7 +198,38 @@ pub fn sql_value_to_value(
         }
         SqlValue::HexStringLiteral(s) => parse_hex_string(s)?,
         SqlValue::Placeholder(s) => return InvalidSqlValueSnafu { value: s }.fail(),
-        _ => todo!("Other sql value"),
+
+        // TODO(dennis): supports binary string
+        _ => {
+            return ConvertSqlValueSnafu {
+                value: sql_val.clone(),
+                datatype: data_type.clone(),
+            }
+            .fail()
+        }
+    })
+}
+
+pub fn value_to_sql_value(val: &Value) -> Result<SqlValue> {
+    Ok(match val {
+        Value::Int8(v) => SqlValue::Number(format!("{}", v), false),
+        Value::UInt8(v) => SqlValue::Number(format!("{}", v), false),
+        Value::Int16(v) => SqlValue::Number(format!("{}", v), false),
+        Value::UInt16(v) => SqlValue::Number(format!("{}", v), false),
+        Value::Int32(v) => SqlValue::Number(format!("{}", v), false),
+        Value::UInt32(v) => SqlValue::Number(format!("{}", v), false),
+        Value::Int64(v) => SqlValue::Number(format!("{}", v), false),
+        Value::UInt64(v) => SqlValue::Number(format!("{}", v), false),
+        Value::Float32(v) => SqlValue::Number(format!("{}", v), false),
+        Value::Float64(v) => SqlValue::Number(format!("{}", v), false),
```

Comment:
nitpick: to_string is more concise.
```suggestion
        Value::Int8(v) => SqlValue::Number(v.to_string(), false),
        Value::UInt8(v) => SqlValue::Number(v.to_string(), false),
        Value::Int16(v) => SqlValue::Number(v.to_string(), false),
        Value::UInt16(v) => SqlValue::Number(v.to_string(), false),
        Value::Int32(v) => SqlValue::Number(v.to_string(), false),
        Value::UInt32(v) => SqlValue::Number(v.to_string(), false),
        Value::Int64(v) => SqlValue::Number(v.to_string(), false),
        Value::UInt64(v) => SqlValue::Number(v.to_string(), false),
        Value::Float32(v) => SqlValue::Number(v.to_string(), false),
        Value::Float64(v) => SqlValue::Number(v.to_string(), false),
```